views/userPage.html

```diff
@@ -52,338 +52,113 @@
-              sourceText = "Because you like "+relatedToBand;
-              break;
-              default:
-              break;
-            }
-            html += '<h3 class="source">'+sourceText+'</h3>'
-          }
-
-          // Adding venue and time
-          html += '<h3 class="venue">'+bandData[i][1]+ ' '+bandData[i][2]+'</h3>';
-
-          // Adding price
-          html += '<h3 class="price">'+bandData[i][4]+'</h3>'
-
-          html += '</a>';
+      // Function that adds an event handler for each "Save to calendar button". index tells us which date and show the "Save to calendar" button was clicked for.
+      function addCalendarButtonEventHandler(index, name, venue, date, time, link){
+        var fullDate = Date.parse(date)
+        var year = fullDate.getFullYear()
+        var month = fullDate.getMonth()+1
+        var date = fullDate.getDate()
+        var hour = Date.parse(time).getHours()
+        var min = Date.parse(time).getMinutes()
+        var endHour;
+        if(hour==23){
+          endHour = 0
+        } else {
+          endHour = hour+1
         }
-        return html;
-      }
-
-      // Function that returns the correct suffix for a given date, e.g. if you input 22, it will return "nd" to form "22nd". The approach here is to extract the second digit from the date (e.g. if you input 23, it'll just look at 3), and return a string based on the value of the second digit: 1 = "st", 2 = "nd", 3 = "rd", everything else = "th".
 
-      function getDateSuffix(date){
-        // Pull the second digit
-        var secondDigit = date % 10
+        var startDateTime = year+"-"+month+"-"+date+"T"+hour+":"+min+":00"
+        var endDateTime = year+"-"+month+"-"+date+"T"+endHour+":"+min+":00"
 
-        // Special cases: 11, 12 and 13 are treated as "th"
-        if(date==11 || date==12 || date==13){
-          return "th"
+        var eventData = {
+          'summary': 'SXSW: '+name,
+          'location': venue,
+          'description': link,
+          'start': startDateTime,
+          'end': endDateTime
         }
-        // 1st, 2nd, 3rd, 4th, 21st, 22nd, 23rd
-        switch(secondDigit){
-          case 1:
-          return "st"
-          break;
-          case 2:
-          return "nd"
-          break;
-          case 3:
-          return "rd"
-          break;
-          default:
-          return "th"
-        }
-      }
 
-      function dayButtonClicked(type){
-        return function(){
-          switch(type){
-            case "previous":
-            if(currentDay==0){
-              currentDay = days.length-1
-            } else {
-              currentDay-=1
+        $("#saveToCalendar-"+index).click(function(e){
+          console.log("Save to calendar button clicked!")
+          console.log("Saving an event at:\n"+eventData.start)
+          e.preventDefault();
+          $.ajax({
+            type: 'POST',
+            url: 'http://localhost:5000/calendar',
```

Comment:
@mathewantony31 we'll need to change this to be a relative URL instead of a fixed one. This currently errors out in production